neurovault/apps/statmaps/tasks.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import absolute_import
+
+import os
+import numpy
+import pandas as pd
+from celery import shared_task
+import nibabel as nib
+from nilearn.plotting import plot_glass_brain
+from nilearn.image import resample_img
+from neurovault.apps.statmaps.models import Image
+
+@shared_task
+def generate_glassbrain_image(nifti_file,pk):
+  nifti_file = str(nifti_file)
+  png_img_name = "glass_brain_%s.png" % pk
+  png_img_path = os.path.join(os.path.split(nifti_file)[0],png_img_name)
+  glass_brain = plot_glass_brain(nifti_file)
+  glass_brain.savefig(png_img_path)
+
+
+@shared_task
+def make_correlation_df(resample_dim=[4,4,4],pkl_path="/opt/image_data/matrices/pearson_corr.pkl"):
+  # Get standard space brain
+  reference = os.path.join(os.environ['FREESURFER_HOME'],'subjects', 'fsaverage', 'mri', 'brainmask.nii.gz')
+  public_images = Image.objects.filter(collection__private=False)
+  # Get all image paths
+  image_paths = [image.file.path for image in public_images]
+  images_resamp, reference_resamp = resample_multi_images_ref(image_paths,reference,resample_dim)
+  # Read into pandas data frame
+  corr_df = pd.DataFrame()
+  row_count = 0
+  for image in images_resamp:
+    corr_df[row_count] = image.get_data().flatten()
+    row_count = row_count + 1
+  corr_df = corr_df.corr(method="pearson")
```

Comment:
shouldn't we create an implicit mask that would consist of an intersection of nonzero voxels from both images before calculating the correlation?